core/shared/src/main/scala/scalaz/zio/RefM.scala

```diff
@@ -66,15 +66,15 @@ object RefM extends Serializable {
     update: A => IO[Nothing, (B, A)],
     promise: Promise[Nothing, B]
   ) {
-    final def run(a: A): IO[List[Throwable], A] =
+    final def run(a: A, ref: Ref[A], onDefect: List[Throwable] => IO[Nothing, Unit]): IO[List[Throwable], Unit] =
       interrupted.get.flatMap {
-        case Some(ts) => IO.fail(ts)
+        case Some(ts) => onDefect(ts) *> IO.fail(ts)
```

Comment:
I think this should just be `onDefect(ts)`, as previously we had (after reduction) 
```scala
IO.fail(ts).redeem(onDefect, _) : IO[Nothing, _]
```
The error channel is erased by `onDefect`.

Reply:
Oh right! Makes the code even simpler.

Reply:
Fixed and now `run` returns `IO[Nothing, Unit]`.